statistics/table.go

```diff
@@ -331,39 +334,65 @@ func (t *Table) ColumnEqualRowCount(sc *stmtctx.StatementContext, value types.Da
 
 // GetRowCountByIntColumnRanges estimates the row count by a slice of IntColumnRange.
 func (coll *HistColl) GetRowCountByIntColumnRanges(sc *stmtctx.StatementContext, colID int64, intRanges []*ranger.Range) (float64, error) {
+	var result float64
 	c, ok := coll.Columns[colID]
 	if !ok || c.IsInvalid(sc, coll.Pseudo) {
 		if len(intRanges) == 0 {
 			return 0, nil
 		}
 		if intRanges[0].LowVal[0].Kind() == types.KindInt64 {
-			return getPseudoRowCountBySignedIntRanges(intRanges, float64(coll.Count)), nil
+			result = getPseudoRowCountBySignedIntRanges(intRanges, float64(coll.Count))
+		} else {
+			result = getPseudoRowCountByUnsignedIntRanges(intRanges, float64(coll.Count))
+		}
+		if sc.EnableOptimizerCETrace && ok {
+			CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, intRanges, "Int Column Stats-Pseudo", uint64(result))
 		}
-		return getPseudoRowCountByUnsignedIntRanges(intRanges, float64(coll.Count)), nil
+		return result, nil
 	}
 	result, err := c.GetColumnRowCount(sc, intRanges, coll.Count, true)
+	if sc.EnableOptimizerCETrace {
+		CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, intRanges, "Int Column Stats", uint64(result))
+	}
 	return result, errors.Trace(err)
 }
 
 // GetRowCountByColumnRanges estimates the row count by a slice of Range.
 func (coll *HistColl) GetRowCountByColumnRanges(sc *stmtctx.StatementContext, colID int64, colRanges []*ranger.Range) (float64, error) {
 	c, ok := coll.Columns[colID]
 	if !ok || c.IsInvalid(sc, coll.Pseudo) {
-		return GetPseudoRowCountByColumnRanges(sc, float64(coll.Count), colRanges, 0)
+		result, err := GetPseudoRowCountByColumnRanges(sc, float64(coll.Count), colRanges, 0)
+		if err == nil && sc.EnableOptimizerCETrace && ok {
+			CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, colRanges, "Column Stats-Pseudo", uint64(result))
+		}
+		return result, err
 	}
 	result, err := c.GetColumnRowCount(sc, colRanges, coll.Count, false)
+	if sc.EnableOptimizerCETrace {
+		CETraceRange(sc, coll.PhysicalID, []string{c.Info.Name.O}, colRanges, "Column Stats", uint64(result))
```

Comment:
`Column Stats` is not unified with `Int Column Stats`, better to use `Nonint Column Stats` or just use `Column Stats` in both places.

Reply:
Updated.